### PR TITLE
[DebugInfo] Fix DebugTypeSubrange parameters order

### PR DESCRIPTION
**Pull request by MrSidims**
Count should be the 3rd parameter.

### CODE REVIEW

File: lib/SPIRV/LLVMToSPIRVDbgTran.cpp [hunk 1]
```diff
@@ -815,7 +815,7 @@ SPIRVEntry *LLVMToSPIRVDbgTran::transDbgSubrangeType(const DISubrange *ST) {
                          : getDebugInfoNoneId();
     }
   };
-  for (int Idx = CountIdx; Idx < OperandCount; ++Idx)
+  for (int Idx = 0; Idx < OperandCount; ++Idx)
```
Comment (vmaksimo) [inline lib/SPIRV/LLVMToSPIRVDbgTran.cpp:818]: We can have less than 4 operands, as the last one is optional. Should it be handled some way here and in reverse translation?

Comment (MrSidims) [reply lib/SPIRV/LLVMToSPIRVDbgTran.cpp]: Moved stride to truly optional

**Comment by asudarsa** [comment]
I think we can ignore clang-format fails. https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2077 should fix this in future runs.

Thanks

**Review by asudarsa — APPROVED** [review]
LGTM. Minor nit . Not blocking. Thanks

File: lib/SPIRV/SPIRVToLLVMDbgTran.cpp [hunk 2]
```diff
@@ -556,8 +556,8 @@ DISubrange *
 SPIRVToLLVMDbgTran::transTypeSubrange(const SPIRVExtInst *DebugInst) {
   using namespace SPIRVDebug::Operand::TypeSubrange;
   const SPIRVWordVec &Ops = DebugInst->getArguments();
-  assert(Ops.size() == OperandCount && "Invalid number of operands");
-  std::vector<Metadata *> TranslatedOps(OperandCount, nullptr);
+  assert(Ops.size() >= MinOperandCount && "Invalid number of operands");
```
Comment (LU-JOHN) [inline lib/SPIRV/SPIRVToLLVMDbgTran.cpp:559]: Add MaxOperandCount check.

File: lib/SPIRV/LLVMToSPIRVDbgTran.cpp [hunk 3]
```diff
@@ -815,7 +818,7 @@ SPIRVEntry *LLVMToSPIRVDbgTran::transDbgSubrangeType(const DISubrange *ST) {
                          : getDebugInfoNoneId();
     }
   };
-  for (int Idx = CountIdx; Idx < OperandCount; ++Idx)
+  for (int Idx = 0; Idx < MaxOperandCount; ++Idx)
     TransOperand(Idx);
   return BM->addDebugInfo(SPIRVDebug::TypeSubrange, getVoidTy(), Ops);
```
Comment (LU-JOHN) [inline lib/SPIRV/LLVMToSPIRVDbgTran.cpp:823]: Consider removing any special treatment of StrideIdx in TransOperand and just resizing to MinOperandCount before the return if Ops[StrideIdx] == getDebugInfoNoneId().

File: lib/SPIRV/SPIRVToLLVMDbgTran.cpp [hunk 4]
```diff
@@ -556,8 +556,8 @@ DISubrange *
 SPIRVToLLVMDbgTran::transTypeSubrange(const SPIRVExtInst *DebugInst) {
   using namespace SPIRVDebug::Operand::TypeSubrange;
   const SPIRVWordVec &Ops = DebugInst->getArguments();
-  assert(Ops.size() == OperandCount && "Invalid number of operands");
-  std::vector<Metadata *> TranslatedOps(OperandCount, nullptr);
```
Comment (LU-JOHN) [inline lib/SPIRV/SPIRVToLLVMDbgTran.cpp:560]: Add MaxOperandCount check.

File: lib/SPIRV/SPIRVToLLVMDbgTran.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (MrSidims) [reply lib/SPIRV/SPIRVToLLVMDbgTran.cpp]: Applied

File: lib/SPIRV/LLVMToSPIRVDbgTran.cpp [hunk 5]
```diff
@@ -815,7 +818,7 @@ SPIRVEntry *LLVMToSPIRVDbgTran::transDbgSubrangeType(const DISubrange *ST) {
                          : getDebugInfoNoneId();
     }
   };
-  for (int Idx = CountIdx; Idx < OperandCount; ++Idx)
+  for (int Idx = 0; Idx < MaxOperandCount; ++Idx)
     TransOperand(Idx);
```
[the remark this answers is not in the document]
Comment (MrSidims) [reply lib/SPIRV/LLVMToSPIRVDbgTran.cpp:816]: Considered, applied

**Review by LU-JOHN — APPROVED** [review]
LGTM

**Pull request merged**